OCCommunicationLib/OCCommunicationLib/Utils/UtilsFramework.m

```diff
@@ -629,10 +629,13 @@ + (BOOL) isServerVersion:(NSString *) serverVersionString higherThanLimitVersion
 
 #pragma mark - Share Permissions
 
-+ (NSInteger) getPermissionsValueByCanEdit:(BOOL)canEdit andCanCreate:(BOOL)canCreate andCanChange:(BOOL)canChange andCanDelete:(BOOL)canDelete andCanShare:(BOOL)canShare andIsFolder:(BOOL) isFolder {
-    
-    NSInteger permissionsValue = k_read_share_permission;
++ (NSInteger) getPermissionsValueByCanRead:(BOOL) canRead andCanEdit:(BOOL)canEdit andCanCreate:(BOOL)canCreate andCanChange:(BOOL)canChange andCanDelete:(BOOL)canDelete andCanShare:(BOOL)canShare andIsFolder:(BOOL) isFolder {
```

Comment:
___I'm going to add this as a comment and not as a change request because I don't know if this is out of scope for this PR but..___

This method definition in the header is +255 characters long and it has a lot of parameters, I think a better approach here could be something similar to:

Create an enumerate for the share options like:
```objective-c
typedef NS_OPTIONS(NSUInteger, OCSharePermission) {
	OCSharePermissionCanRead = 1<<0,
	OCSharePermissionCanUpdate = 1<<1,
	OCSharePermissionCanCreate = 1<<2,
	OCSharePermissionCanDelete = 1<<3,
	OCSharePermissionCanShare = 1<<4
};
```

Then you change all those arguments with a more clear enumeration parameter like: 
```objective-c
+ (NSInteger)getSharePermissonValueBy:(OCSharePermission)permissions withFolder:(BOOL)isFolder {
	NSInteger permissionsValue = 0;

	if (isFolder) {

		if (permissions & OCSharePermissionCanCreate) {
			permissionsValue += k_create_share_permission;
		}

		if (permissions & OCSharePermissionCanUpdate) {
			permissionsValue += k_update_share_permission;
		}

		if (permissions & OCSharePermissionCanDelete) {
			permissionsValue += k_delete_share_permission;
		}

	} else {

		if (permissions & OCSharePermissionCanUpdate) {
			permissionsValue += k_update_share_permission;
		}
	}

	if (permissions & OCSharePermissionCanRead) {
		permissionsValue += k_read_share_permission;
	}

	if (permissions & OCSharePermissionCanShare) {
		permissionsValue += k_share_share_permission;
	}

	return permissionsValue;
}
```

This leads to a better and cleaner call like:
```objective-c
[self getSharePermissonValueBy: (OCSharePermissionCanDelete | OCSharePermissionCanUpdate | OCSharePermissionCanCreate) withFolder:YES];
```